src/features/elementfeature.jl

```diff
@@ -74,27 +94,48 @@ end
 
 function (f::ElementFeatureDescriptor)(a::AbstractAtoms)
     @assert all([el in encodable_elements(f) for el in a.elements]) "Feature $(f.name) cannot encode some element(s) in this structure!"
+    f.encoder_decoder(f, a, ENCODE)
+end
+
+
+function (ed::DummyED)(e::ElementFeatureDescriptor, a::AbstractAtoms, e_or_d::EncodeOrDecode)
+    if e_or_d == ENCODE
+        ed.encode_f(e, a, ed.nbins, ed.logspaced)
+    else
+        ed.decode(e, a, ed.nbins, ed.logspaced)
+    end
+end
+
+decode(f::ElementFeatureDescriptor, encoded_feature) =
+    f.encoder_decoder.decode_f(f, encoded_feature)
+
+function default_efd_encode(
+    f::ElementFeatureDescriptor,
+    a::AbstractAtoms,
+    nbins::Integer,
+    logspaced::Bool,
+)
     reduce(
         hcat,
         map(
             e -> onehot_lookup_encoder(
                 e,
                 f.name,
                 f.lookup_table;
-                nbins = f.nbins,
-                logspaced = f.logspaced,
+                nbins,
+                logspaced,
                 categorical = f.categorical,
             ),
             a.elements,
         ),
     )
 end
 
-decode(f::ElementFeatureDescriptor, encoded_feature) = onecold_decoder(
+default_efd_decode(e::ElementFeatureDescriptor, encoded_feature) = onecold_decoder(
     encoded_feature,
-    f.name,
-    f.lookup_table;
-    nbins = f.nbins,
-    logspaced = f.logspaced,
-    categorical = f.categorical,
+    e.name,
+    e.lookup_table;
+    e.encoder_decoder.nbins,
```

Comment:
is there a reason that this one pulls these two arguments directly from `e.encoder_decoder` whereas `default_efd_encode` requires them to be directly provided?

Reply:
Resolved in 454574d.